docs/devel/node-performance-testing.md

```diff
@@ -0,0 +1,146 @@
+
+There are lots of factors which can affect node performance numbers, so care must be taken in
+setting up the cluster to make the intended measurements. In addition to taking the following steps
+into consideration, it is important to document precisely which setup was used. For example,
+performance can vary wildly from commit-to-commit, so it is very important to document which commit
+or version of Kubernetes was used, which Docker version was used, etc.
+
+### Addon pods
+
+Be aware of which addon pods are running on which nodes. By default Kubernetes runs 8 addon pods,
+plus another 2 per node (`fluentd-elasticsearch` and `kube-proxy`) in the `kube-system`
+namespace. The addon pods can be disabled for more consistent results, but doing so can also have
+performance implications.
+
+For example, Heapster polls each node regularly to collect stats data. Disabling Heapster will hide
+the performance cost of serving those stats in the Kubelet.
+
+#### Disabling Add-ons
+
+Disabling addons is simple. Just ssh into the Kubernetes master and move or delete the addon from
+`/etc/kubernetes/addons/`.
+
+### Which / how many pods?
+
+Performance will vary a lot between a node with 0 pods and a node with 100 pods. In many cases
+you'll want to make measurements with several different amounts of pods. Scaling a replication
+controller makes this easy, just make sure the system reaches a steady-state before starting the
+measurement. E.g. `kubectl scale replicationcontroller pause --replicas=100`
+
+In most cases pause pods will yield the most consistent measurements since the system will not be
+affected by pod load. However, in some special cases Kubernetes has been tuned to optimize pods that
+are not doing anything, such as the cAdvisor housekeeping (stats gathering). In these cases,
+performing a very light task (such as a simple network ping) can make a difference.
+
+Finally, you should also consider which features yours pods should be using. For example, if you
+want to measure performance with probing, you should obviously use pods with liveness or readiness
+probes configured. Likewise for volumes, number of containers, etc.
+
+### Other Tips
+
+**Number of nodes** - On the one hand, it can be easier to manage logs, pods, environment etc. with
+  a single node to worry about. On the other hand, having multiple nodes will let you gather more
+  data in parallel for more robust sampling.
+
+## E2E Performance Test
+
+There is an end-to-end test for collecting overall resource usage of node components:
+[kubelet_perf.go](../../test/e2e/kubelet_perf.go). To
+run the test, simply make sure you have an e2e cluster running (`go run hack/e2e.go -up`) and
+[set up](#cluster-set-up) correctly.
+
+Run the test with `go run hack/e2e.go -v -test
+--test_args="--ginkgo.focus=resource\susage\stracking`. You may also with to customise the number of
+pods or other elements of the test (remember to rerun `make WHAT=test/e2e/e2e.test` after you do).
+
```

Comment:
Maybe add a section about prometheus or just have a link pointing to a doc? I couldn't find any useful esiting doc other than https://github.com/kubernetes/kubernetes/blob/master/docs/devel/instrumentation.md though